docs/source/app-dev/command-deduplication.rst

```diff
@@ -0,0 +1,352 @@
+In particular, the submission may be a duplicate even if the command submission service acknowledges the submission with the gRPC status code ``OK``.
+
+- If no conflicting submission with the same :ref:`change ID <change-id>` was found, the completion event and possibly the response convey the result of the submission (success or a gRPC error).
+
+- The gRPC status code ``ALREADY_EXISTS`` with error code ID :ref:`DUPLICATE_COMMAND <error_code_DUPLICATE_COMMAND>` if there is an earlier command completion for the same :ref:`change ID <change-id>` within the effective deduplication period.
+
+- The gRPC status code ``ABORTED`` with error code id :ref:`SUBMISSION_ALREADY_IN_FLIGHT <error_code_SUBMISSION_ALREADY_IN_FLIGHT>` if another submission for the same :ref:`change ID <change-id>` was in flight when this submission was processed.
+
+- The gRPC status code ``FAILED_PRECONDITION`` with error code id :ref:`INVALID_DEDUPLICATION_PERIOD <error_code_INVALID_DEDUPLICATION_PERIOD>` if the specified deduplication period is not supported.
+  The fields ``longest_duration`` or ``earliest_offset`` in the metadata specify the longest duration or earliest offset that is currently supported on the Ledger API endpoint.
+  At least one of the two fields is present.
+
+  Neither deduplication durations up to the :ref:`maximum deduplication time <com.daml.ledger.api.v1.LedgerConfiguration.max_deduplication_time>` nor deduplication offsets published within that time SHOULD not result in this error.
+  Participants may accept longer periods at their discretion.
+
+For deduplication to work as intended, all submissions for the same ledger change must be submitted via the same participant.
+This is because a participant outputs by default only the completion events for submissions that were requested via the very same participant,
+and whether a submission is considered a duplicate is determined by the completion events.
+
+On some ledgers, every participant outputs the completion events for all the hosted parties, not just for the submissions that went through the participant.
+In this case, command deduplication works across participants.
+At this time, only `Daml Driver for VMware Blockchain <https://www.digitalasset.com/daml-for-vmware-blockchain/>`__ supports command deduplication across participants.
+
+    
+
+How to use command deduplication
+********************************
+
+When an application wants to effectuate a ledger change exactly once, the application must resubmit a command if an earlier submission was lost.
+However, the application typically cannot distinguish a lost submission from a Daml ledger that is just slow in processing the submissions.
+So the application should just resubmit the command until it is executed against the ledger, and rely on command deduplication to reject all duplicate submissions that do make it to the ledger.
```

Comment:
"Command deduplication allows the application to resubmit the command until it is executed and reject all duplicate submissions thereafter."

Reply:
Changed. I feel that there's some subtlety lost, namely that an earlier (from the application perspective) submission may also be rejected by command deduplication, namely if it was overtaken by a later submission on its way to the participant or while the participant was processing it. But that's probably too much detail here.